python/tank/bootstrap/manager.py

```diff
@@ -516,7 +516,21 @@ def prepare_engine(self, engine_name, entity):
 
         return path
 
-    def get_pipeline_configurations(self, project):
+    def get_resolved_pipeline_configuration_descriptor(self, project_id=None):
+        """
+        Resolves a pipeline configuration and returns its associated descriptor object.
```

Comment:
Correct. We're in a situation where something like tk-config-basic could (and likely will) be used by multiple projects. In that situation we can reuse the already-cached data, even if it wasn't cached by the current project.